### PR TITLE
Fix MB-1925

### PR DESCRIPTION
**Pull request by a5anka**
The issue is fixed by using AMQChannel's ID for transaction association.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java [hunk 1]
```diff
@@ -219,7 +218,7 @@ boolean disassociateSession(long sessionID) {
      * @param sessionID session identifier of the session
      * @return True if there is a matching suspended entry
      */
-    boolean resumeSession(long sessionID) {
+    boolean resumeSession(UUID sessionID) {
```
Comment (pamod) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java]: How about sticking with the convention of "sessionId" in camelcase ?

**Comment by Asitha** [comment]
Shall we add a test case for this?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
     /**
      * Session Id of a session that is recovered from storage
      */
-    public static final int RECOVERY_SESSION_ID = -1;
+    public static final UUID RECOVERY_SESSION_ID = UUID.randomUUID();
```
Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java]: Can you explain how a random value replaces the previous constant value?

Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java]: Actually, we don't need this constant. It is used as the session id for recovered branches.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java (hunk 1 above)
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java]: :+1:

**Pull request merged**